document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/service/impl/DocumentGeneratorServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @Autowired
+    public DocumentGeneratorServiceImpl(DocumentInfoService documentInfoService, EnvironmentReader environmentReader,
+                                        RenderDocumentRequestHandler requestHandler) {
+
+        this.documentInfoService = documentInfoService;
+        this.environmentReader = environmentReader;
+        this.requestHandler = requestHandler;
+    }
+
+    @Override
+    public DocumentResponse generate(DocumentRequest documentRequest) {
+
+        DocumentInfo documentInfo = null;
+        RenderDocumentResponse renderResponse;
+        DocumentResponse response = null;
+
+        //TODO addition of get doc gen type from URL to be added in SFA 580
+
+        try {
+            //TODO currently no impl present, being completed in SFA 567
+            documentInfo = documentInfoService.getDocumentInfo();
+        } catch (Exception e){
+            LOG.error(e);
+        }
+
+        if (documentInfo != null) {
+            renderResponse = renderSubmittedDocumentData(documentRequest, documentInfo);
+            response = setDocumentResponse(renderResponse, documentInfo);
+        }
+
+        if (response != null) {
+            return response;
+        }
+
+        return null;
+    }
+
+    /**
+     * Send data to Render Service and generate document
+     *
+     * @param documentRequest
+     * @param documentInfo
+     * @return
+     */
+    private RenderDocumentResponse renderSubmittedDocumentData(DocumentRequest documentRequest,
+                                                               DocumentInfo documentInfo) {
+
+        String host = environmentReader.getMandatoryString(DOCUMENT_RENDER_SERVICE_HOST_ENV_VAR);
+        String url = host + CONTEXT_PATH;
+
+        RenderDocumentRequest requestData = new RenderDocumentRequest();
+        requestData.setAssetId(documentInfo.getAssetId());
+        requestData.setContentType(documentRequest.getContentType());
+        requestData.setData(documentInfo.getData());
+        requestData.setDocumentType(documentRequest.getDocumentType());
+        requestData.setTemplateName(documentInfo.getTemplateName());
+        requestData.setLocation(documentInfo.getLocation());
+
+        try {
+            return requestHandler.sendDataToDocumentRenderService(url, requestData);
+        } catch (Exception e) {
+            LOG.error(e);
+        }
+
+        return null;
+    }
+
+    /**
+     * Set documentResponse for Api
+     *
+     * @param renderResponse
+     * @param documentInfo
+     * @return
+     */
+    private DocumentResponse setDocumentResponse(RenderDocumentResponse renderResponse, DocumentInfo documentInfo) {
+
+        DocumentResponse response;
+
+        response = new DocumentResponse();
+        response.setLocation(renderResponse != null ? renderResponse.getLocation() : null);
```

Comment:
Can we have a `renderResponse != null` check wrapping these setters? That way the values will just remain null rather than being explicitly set as null

Reply:
what's wrong with explicitly setting them to ensure they are null? It is important that they are. Don't want any instance of them not being.